Doc/reference/toplevel_components.rst

```diff
@@ -66,7 +66,9 @@ File input
 
 All input read from non-interactive files has the same form:
 
-.. productionlist:: python-grammar
+.. grammar-snippet::
+   :group: python-grammar
```

Comment:
Why `:group:` instead of a directive argument?

Reply:
Good question :)

I'd eventually like to use directive arguments for listing the rules to be documented (here, `file_input`), with the body of the directive being autogenerated or at least checked.
I could keep using arguments for group instead, but eventually I'd add *some* option, which means requiring an empty line.

The changed syntax also makes it easier to see if it's the new directive or the old one.

Does that make sense? I can change it back, of course.

Reply:
Perhaps we could use that argument now? Even if unused, it shows what the intention is.

P.S. you probably don't want `final_argument_whitespace = True`, that's often for when we merge parsing of directive arguments and content (e.g. admonitions can be used with or without a blank line).

Reply:
(resolved "offline")